Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs

```diff
@@ -98,19 +103,10 @@ private void RefreshSolverHandler()
             }
         }
 
-        private void DestroySolver()
+        private void DisableSolver()
```

Comment:
I can see the reasons for the rename, but these have been bulked together in a related change.  If these are to be renamed, there should be a corresponding issue request to discuss the change and a separate PR to address it.

Each PR should change specifically only what it is intended to address or we lose tracability.

Reply:
The naming change is fine by me - the intent here is to provide a change along with the qualifying stuff that demonstrates that it works. I don't think that the magnitude of the changes here warrant having a split even though I generally agree with the principle of splitting things up into smaller changes.

---
In reply to: [314231637](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5662#discussion_r314231637) [](ancestors = 314231637)

Reply:
I'm fine with this change being included but I did DestroyImmediate on purpose for a different reason.

In a real application, re-using resources and not wasting cycles on GarbageCollection makes sense. But this is designed to be an example scene to learn from. 

The solvers already have a crap load of properties to understand. I preferred only showing one solver at a time on the gameobject so it is very clear to the new developer. 

The idea being is you could select a solver and then in editor select that gameobject and modify the properties to test things out. Even disabled, having possibly 3-5 solvers or more as we grow, clutters up that inspector

Reply:
> But this is designed to be an example scene to learn from.

Though Unity Docs clearly state that they recommend to use `Destroy` instead of `DestroyImmediate`, as `DestroyImmediate` is intended for Editor Scripts only. So this would show learners a wrong use of Unity API.

The learning argument could also be used, to show them how they can simply switch between different Solvers, without doing complicated component destroy and rebuild procedures. But instead going the natural Unity was of simply disabling components that are temporarily not used.

Reply:
I’m advocating destroying the components, not the manner in which they are destroyed. 

Switching between solvers dynamically is an edge case of common developer usage. The larger reason for the example scene is to try the solvers out in scene view and play with each solver’a properties. Honestly, the solver scene description and UX could heavily be improved all up